File: java/0001-two-sum/Solution.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> diffIdx = new HashMap<>();

        for (int i = 0; i < nums.length; ++i) {
            if (diffIdx.containsKey(nums[i])) {
                return new int[] { i, diffIdx.get(nums[i]) };
            } else {
                diffIdx.put(target - nums[i], i);
            }
        }

        return null;
    }
}

File: c++/0001-two-sum/solution.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> diffIdx;

        for (int i = 0; i < nums.size(); ++i) {
            if (diffIdx.find(nums[i]) != diffIdx.end()) {
                return {i, diffIdx[nums[i]]};
            } else {
                diffIdx[target - nums[i]] = i;
            }
        }

        return {};
    }
};
